File: src/test/java/name/gyger/jmoney/EntityFactory.java
package name.gyger.jmoney;

import name.gyger.jmoney.account.Account;
import name.gyger.jmoney.account.AccountService;
import name.gyger.jmoney.account.Entry;
import name.gyger.jmoney.account.EntryService;
import name.gyger.jmoney.category.Category;
import name.gyger.jmoney.category.CategoryService;

import java.util.stream.IntStream;

public class EntityFactory {

    public static void createEntries(Entry entry, int count, EntryService entryService) {
        IntStream.range(0, count).forEach(i -> {
            entryService.createEntry(entry);
        });
    }

    public static long createAccount(String name, long startBalance, AccountService accountService) {
        Account account = new Account();
        account.setName(name);
        account.setStartBalance(startBalance);
        return accountService.createAccount(account);
    }

    public static long createTopLevelCategory(String name, CategoryService categoryService) {
        Category catA = new Category();
        catA.setName(name);
        catA.setParentId(categoryService.getRootCategory().getId());
        return categoryService.createCategory(catA);
    }

}

File: src/test/kotlin/name/gyger/jmoney/EntityFactory.kt
package name.gyger.jmoney

import name.gyger.jmoney.account.Account
import name.gyger.jmoney.account.AccountService
import name.gyger.jmoney.account.Entry
import name.gyger.jmoney.account.EntryService
import name.gyger.jmoney.category.Category
import name.gyger.jmoney.category.CategoryService

import java.util.stream.IntStream

fun createEntries(entryProto: Entry, count: Int, entryService: EntryService) {
    IntStream.range(0, count).forEach {
        val entry = Entry()
        entry.accountId = entryProto.accountId
        entry.categoryId = entryProto.categoryId
        entry.date = entryProto.date
        entry.amount = entryProto.amount
        entryService.deepSave(entry)
    }
}

fun createAccount(name: String, startBalance: Long, accountService: AccountService): Long {
    val account = Account()
    account.name = name
    account.startBalance = startBalance
    return accountService.createAccount(account)
}

fun createTopLevelCategory(name: String, categoryService: CategoryService): Long {
    val catA = Category()
    catA.name = name
    catA.parentId = categoryService.getRootCategory().id
    return categoryService.createCategory(catA)
}

